default.nix

```diff
@@ -0,0 +1,218 @@
+      additionBottom
+      additionTop
+      generationInstructions
+      (shareTable 29 24 13 25)
+      (shareTable 9 8 23 18)
+      (shareTable 22 31 27 19)
+      (shareTable 1 0 3 16)
+      (shareTable 11 28 12 14)
+      (shareTable 6 4 2 15)
+      (shareTable 10 17 21 20)
+      (shareTable 26 30 7 5)
+      recovery
+      fusionInner
+      fusionOuter
+      checksumTable1
+      checksumTable2
+      checksumWorksheet
+    ];
+  };
+
+  dependencyContentRecur = content: builtins.concatMap
+    (item: (dependencyContentRecur item.dependencies) ++ [ item.content ])
+    content;
+  dependencyContent = pages: lib.lists.unique
+    (builtins.concatMap (page: dependencyContentRecur page.dependencies) pages);
+
+  renderBooklet = booklet:
+  let
+    addPage = content: pageData: {
+      content = content.content + lib.optionalString (pageData ? sourceHeader) ''
+        %****************************************************************
+        %*
+        %* ${pageData.sourceHeader}
+        %*
+        %****************************************************************
+      '' + ''
+        %%Page: ${toString content.nextPgIdx} ${toString content.nextPgIdx}
+        ${lib.optionalString (pageData ? isLandscape) "%%PageOrientation: Landscape\n"}%%BeginPageSetup
+        /pgsave save def
+        %%EndPageSetup
+      '' + (
+        if pageData ? drawPageContent
+        then
+          if pageData ? isLandscape
+          then "landscapePage begin ${toString content.nextCtIdx} drawPageContent\n"
+          else "portraitPage begin ${toString content.nextCtIdx} drawPageContent\n"
+        else
+          if pageData ? isLandscape
+          then "90 rotate\n"
+          else ""
+      ) + ''
+        ${pageData.content}
+        ${lib.optionalString (pageData ? drawPageContent) "end\n"}pgsave restore
+        showpage
+      '';
+      nextCtIdx = content.nextCtIdx + (if pageData ? drawPageContent then 1 else 0);
+      nextPgIdx = content.nextPgIdx + 1;
+    };
+    initialContent = {
+      content = ''
+        %!PS-Adobe-3.0
+        %%Orientation: Portrait
+        %%Pages: ${toString (builtins.length booklet.pages)}
+        %%EndComments
+        %%BeginSetup
+        ${toString (dependencyContent (booklet.pages))}%%EndSetup
+
+        %************************************************************************
+        %************************************************************************
+        %*
+        %* Section Three: Page Rendering
+        %*
+        %************************************************************************
+        %************************************************************************
+
+      '';
+      nextPgIdx = 1;
+      nextCtIdx = 1;
+    };
+    finalContent = builtins.foldl' addPage initialContent booklet.pages;
+  in
+    pkgs.writeTextFile {
+      name = booklet.name;
+      text = finalContent.content + ''
+        %%EOF
+      '';
+    };
+in
+stdenv.mkDerivation {
```

Comment:
Why not just put renderBooklet as the derivation result?  Wouldn't that make it easier to check the result into git?  We can do `nix-build -o SSS32.ps` and then commit SSS32.ps. (I haven't actually tried this).

Reply:
Because I eventually want to produce more than one output -- the full booklet, but also postscript files with individual worksheets, 256-bit vesrions of the worksheets, etc. And maybe a illustrated version alongside the regular version.

Reply:
On further reflection, and having learned Nix better, let me explore making this derivation have multiple outputs, where the default is the fully-assembled SSS32.ps. I'm unsure if this is the right way to go still, because there are e.g. flags we want to be able to set about PDF vs PS, color vs non-color etc, and it might still be the cleanest thing to output a directory with a bunch of different files.

Reply:
My experiments with git and symlinks was a failure.  While we could support multiple outputs, I do not think that it worth worrying about.  At least not at this time.